code/game/objects/items/weapons/flamethrower.dm

```diff
@@ -69,7 +70,41 @@
 			flame_turf(turflist)
 
 /obj/item/weapon/flamethrower/attackby(obj/item/W, mob/user, params)
-	if(istype(W,/obj/item/weapon/tank/internals/plasma))
+	if(istype(W, /obj/item/weapon/wrench) && !status)//Taking this apart
+		var/turf/T = get_turf(src)
+		if(weldtool)
+			weldtool.loc = T
+			weldtool = null
+		if(igniter)
+			igniter.loc = T
+			igniter = null
+		if(ptank)
+			ptank.loc = T
+			ptank = null
+		new /obj/item/stack/rods(T)
+		qdel(src)
+		return
+
+	else if(istype(W, /obj/item/weapon/screwdriver) && igniter && !lit)
+		status = !status
+		user << "<span class='notice'>[igniter] is now [status ? "secured" : "unsecured"]!</span>"
+		update_icon()
+		return
+
+	else if(isigniter(W))
```

Comment:
because this is the kind of garbage code you need when your crafting system is based on attackbys!